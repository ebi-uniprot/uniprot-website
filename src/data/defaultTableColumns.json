["accession", "id", "protein_name", "gene_names", "organism"]
