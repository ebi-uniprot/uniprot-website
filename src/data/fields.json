{
  "data": [
    {
      "id": "Date of",
      "title": "Date of",
      "items": [
        {
          "id": "date_create",
          "label": "Date of creation"
        },
        {
          "id": "date_mod",
          "label": "Date of last modification"
        },
        {
          "id": "date_seq_mod",
          "label": "Date of last sequence modification"
        },
        {
          "id": "version",
          "label": "Entry version"
        }
      ]
    },
    {
      "id": "Expression",
      "title": "Expression",
      "items": [
        {
          "id": "cc:developmental_stage",
          "label": "Developmental stage"
        },
        {
          "id": "cc:induction",
          "label": "Induction"
        },
        {
          "id": "cc:tissue_specificity",
          "label": "Tissue specificity"
        }
      ]
    },
    {
      "id": "Family & Domains",
      "title": "Family & Domains",
      "items": [
        {
          "id": "ft:coiled",
          "label": "Coiled coil"
        },
        {
          "id": "ft:compbias",
          "label": "Compositional bias"
        },
        {
          "id": "cc:domain",
          "label": "Domain [CC]"
        },
        {
          "id": "ft:domain",
          "label": "Domain [FT]"
        },
        {
          "id": "ft:motif",
          "label": "Motif"
        },
        {
          "id": "protein_families",
          "label": "Protein families"
        },
        {
          "id": "ft:region",
          "label": "Region"
        },
        {
          "id": "ft:repeat",
          "label": "Repeat"
        },
        {
          "id": "cc:similarity",
          "label": "Sequence similarities"
        },
        {
          "id": "ft:zn_fing",
          "label": "Zinc finger"
        }
      ]
    },
    {
      "id": "Function",
      "title": "Function",
      "items": [
        {
          "id": "absorption",
          "label": "Absorption"
        },
        {
          "id": "ft:act_site",
          "label": "Active site"
        },
        {
          "id": "ft:binding",
          "label": "Binding site"
        },
        {
          "id": "ft:ca_bind",
          "label": "Calcium binding"
        },
        {
          "id": "cc:catalytic_activity",
          "label": "Catalytic activity"
        },
        {
          "id": "cc:cofactor",
          "label": "Cofactor"
        },
        {
          "id": "ft:dna_bind",
          "label": "DNA binding"
        },
        {
          "id": "ec",
          "label": "EC number"
        },
        {
          "id": "cc:enzyme_regulation",
          "label": "Enzyme regulation"
        },
        {
          "id": "cc:function",
          "label": "Function [CC]"
        },
        {
          "id": "kinetics",
          "label": "Kinetics"
        },
        {
          "id": "ft:metal",
          "label": "Metal binding"
        },
        {
          "id": "ft:np_bind",
          "label": "Nucleotide binding"
        },
        {
          "id": "cc:pathway",
          "label": "Pathway"
        },
        {
          "id": "ph_dependence",
          "label": "pH dependence"
        },
        {
          "id": "redox_potential",
          "label": "Redox potential"
        },
        {
          "id": "ft:site",
          "label": "Site"
        },
        {
          "id": "temp_dependence",
          "label": "Temperature dependence"
        }
      ]
    },
    {
      "id": "Gene Ontology (GO)",
      "title": "Gene Ontology (GO)",
      "items": [
        {
          "id": "go_p",
          "label": "Gene Ontology (biological process)"
        },
        {
          "id": "go_c",
          "label": "Gene Ontology (cellular component)"
        },
        {
          "id": "go",
          "label": "Gene Ontology (GO)"
        },
        {
          "id": "go_f",
          "label": "Gene Ontology (molecular function)"
        },
        {
          "id": "go_id",
          "label": "Gene Ontology IDs"
        }
      ]
    },
    {
      "id": "Interaction",
      "title": "Interaction",
      "items": [
        {
          "id": "cc:interaction",
          "label": "Interacts with"
        },
        {
          "id": "cc:subunit",
          "label": "Subunit structure"
        }
      ]
    },
    {
      "id": "Miscellaneous",
      "title": "Miscellaneous",
      "items": [
        {
          "id": "score",
          "label": "Annotation"
        },
        {
          "id": "cc:caution",
          "label": "Caution"
        },
        {
          "id": "feature",
          "label": "Features"
        },
        {
          "id": "keyword",
          "label": "Keywords"
        },
        {
          "id": "keywordid",
          "label": "Keyword ID"
        },
        {
          "id": "matched_text",
          "label": "Matched text"
        },
        {
          "id": "cc:miscellaneous",
          "label": "Miscellaneous [CC]"
        },
        {
          "id": "protein_existence",
          "label": "Protein existence"
        },
        {
          "id": "reviewed",
          "label": "Reviewed"
        },
        {
          "id": "tools",
          "label": "Tools"
        },
        {
          "id": "uniparc_id",
          "label": "UniParc"
        }
      ]
    },
    {
      "id": "Names & Taxonomy",
      "title": "Names & Taxonomy",
      "items": [
        {
          "id": "accession",
          "label": "Entry"
        },
        {
          "id": "id",
          "label": "Entry Name"
        },
        {
          "id": "gene_names",
          "label": "Gene Names"
        },
        {
          "id": "gene_oln",
          "label": "Gene Names (ordered locus)"
        },
        {
          "id": "gene_orf",
          "label": "Gene Names (ORF)"
        },
        {
          "id": "gene_primary",
          "label": "Gene Names (primary)"
        },
        {
          "id": "gene_synonym",
          "label": "Gene Names (synonym)"
        },
        {
          "id": "organism",
          "label": "Organism"
        },
        {
          "id": "organism_id",
          "label": "Organism (ID)"
        },
        {
          "id": "protein_name",
          "label": "Protein names"
        },
        {
          "id": "dr:proteomes",
          "label": "Proteomes"
        },
        {
          "id": "lineage",
          "label": "Taxonomic lineage"
        },
        {
          "id": "organism_host",
          "label": "Virus hosts"
        }
      ]
    },
    {
      "id": "PTM / Processing",
      "title": "PTM / Processing",
      "items": [
        {
          "id": "ft:chain",
          "label": "Chain"
        },
        {
          "id": "ft:crosslnk",
          "label": "Cross-link"
        },
        {
          "id": "ft:disulfide",
          "label": "Disulfide bond"
        },
        {
          "id": "ft:carbohyd",
          "label": "Glycosylation"
        },
        {
          "id": "ft:init_met",
          "label": "Initiator methionine"
        },
        {
          "id": "ft:lipid",
          "label": "Lipidation"
        },
        {
          "id": "ft:mod_res",
          "label": "Modified residue"
        },
        {
          "id": "ft:peptide",
          "label": "Peptide"
        },
        {
          "id": "cc:ptm",
          "label": "Post-translational modification"
        },
        {
          "id": "ft:propep",
          "label": "Propeptide"
        },
        {
          "id": "ft:signal",
          "label": "Signal peptide"
        },
        {
          "id": "ft:transit",
          "label": "Transit peptide"
        }
      ]
    },
    {
      "id": "Pathology & Biotech",
      "title": "Pathology & Biotech",
      "items": [
        {
          "id": "cc:allergen",
          "label": "Allergenic Properties"
        },
        {
          "id": "cc:biotechnology",
          "label": "Biotechnological use"
        },
        {
          "id": "cc:disruption_phenotype",
          "label": "Disruption phenotype"
        },
        {
          "id": "cc:disease",
          "label": "Involvement in disease"
        },
        {
          "id": "ft:mutagen",
          "label": "Mutagenesis"
        },
        {
          "id": "cc:pharmaceutical",
          "label": "Pharmaceutical use"
        },
        {
          "id": "cc:toxic_dose",
          "label": "Toxic dose"
        }
      ]
    },
    {
      "id": "Publications",
      "title": "Publications",
      "items": [
        {
          "id": "mapped_pm_id",
          "label": "Mapped PubMed ID"
        },
        {
          "id": "pm_id",
          "label": "PubMed ID"
        }
      ]
    },
    {
      "id": "Sequences",
      "title": "Sequences",
      "items": [
        {
          "id": "cc:alternative_products",
          "label": "Alternative products (isoforms)"
        },
        {
          "id": "ft:var_seq",
          "label": "Alternative sequence"
        },
        {
          "id": "error_gmodel_pred",
          "label": "Erroneous gene model prediction"
        },
        {
          "id": "fragment",
          "label": "Fragment"
        },
        {
          "id": "gene_location",
          "label": "Gene encoded by"
        },
        {
          "id": "length",
          "label": "Length"
        },
        {
          "id": "mass",
          "label": "Mass"
        },
        {
          "id": "cc:mass_spectrometry",
          "label": "Mass spectrometry"
        },
        {
          "id": "ft:variant",
          "label": "Natural variant"
        },
        {
          "id": "ft:non_con",
          "label": "Non-adjacent residue"
        },
        {
          "id": "ft:non_std",
          "label": "Non-standard residue"
        },
        {
          "id": "ft:non_ter",
          "label": "Non-terminal residue"
        },
        {
          "id": "cc:polymorphism",
          "label": "Polymorphism"
        },
        {
          "id": "cc:rna_editing",
          "label": "RNA Editing"
        },
        {
          "id": "sequence",
          "label": "Sequence"
        },
        {
          "id": "cc:sequence_caution",
          "label": "Sequence caution"
        },
        {
          "id": "ft:conflict",
          "label": "Sequence conflict"
        },
        {
          "id": "ft:unsure",
          "label": "Sequence uncertainty"
        },
        {
          "id": "sequence_version",
          "label": "Sequence version"
        }
      ]
    },
    {
      "id": "Structure",
      "title": "Structure",
      "items": [
        {
          "id": "3d",
          "label": "3D"
        },
        {
          "id": "ft:strand",
          "label": "Beta strand"
        },
        {
          "id": "ft:helix",
          "label": "Helix"
        },
        {
          "id": "ft:turn",
          "label": "Turn"
        }
      ]
    },
    {
      "id": "Subcellular location",
      "title": "Subcellular location",
      "items": [
        {
          "id": "ft:intramem",
          "label": "Intramembrane"
        },
        {
          "id": "cc:subcellular_location",
          "label": "Subcellular location [CC]"
        },
        {
          "id": "ft:top_dom",
          "label": "Topological domain"
        },
        {
          "id": "ft:transmem",
          "label": "Transmembrane"
        }
      ]
    },
    {
      "id": "Taxonomic identifier",
      "title": "Taxonomic identifier",
      "items": [
        {
          "id": "tax_id",
          "label": "Taxonomic lineage IDs"
        }
      ]
    },
    {
      "id": "Taxonomic lineage",
      "title": "Taxonomic lineage",
      "items": [
        {
          "id": "tl:all",
          "label": "Taxonomic lineage (all)"
        },
        {
          "id": "tl:class",
          "label": "Taxonomic lineage (CLASS)"
        },
        {
          "id": "tl:cohort",
          "label": "Taxonomic lineage (COHORT)"
        },
        {
          "id": "tl:family",
          "label": "Taxonomic lineage (FAMILY)"
        },
        {
          "id": "tl:forma",
          "label": "Taxonomic lineage (FORMA)"
        },
        {
          "id": "tl:genus",
          "label": "Taxonomic lineage (GENUS)"
        },
        {
          "id": "tl:infraclass",
          "label": "Taxonomic lineage (INFRACLASS)"
        },
        {
          "id": "tl:infraorder",
          "label": "Taxonomic lineage (INFRAORDER)"
        },
        {
          "id": "tl:kingdom",
          "label": "Taxonomic lineage (KINGDOM)"
        },
        {
          "id": "tl:order",
          "label": "Taxonomic lineage (ORDER)"
        },
        {
          "id": "tl:parvorder",
          "label": "Taxonomic lineage (PARVORDER)"
        },
        {
          "id": "tl:phylum",
          "label": "Taxonomic lineage (PHYLUM)"
        },
        {
          "id": "tl:species",
          "label": "Taxonomic lineage (SPECIES)"
        },
        {
          "id": "tl:species_group",
          "label": "Taxonomic lineage (SPECIES_GROUP)"
        },
        {
          "id": "tl:species_subgroup",
          "label": "Taxonomic lineage (SPECIES_SUBGROUP)"
        },
        {
          "id": "tl:subclass",
          "label": "Taxonomic lineage (SUBCLASS)"
        },
        {
          "id": "tl:subcohort",
          "label": "Taxonomic lineage (SUBCOHORT)"
        },
        {
          "id": "tl:subfamily",
          "label": "Taxonomic lineage (SUBFAMILY)"
        },
        {
          "id": "tl:subgenus",
          "label": "Taxonomic lineage (SUBGENUS)"
        },
        {
          "id": "tl:subkingdom",
          "label": "Taxonomic lineage (SUBKINGDOM)"
        },
        {
          "id": "tl:suborder",
          "label": "Taxonomic lineage (SUBORDER)"
        },
        {
          "id": "tl:subphylum",
          "label": "Taxonomic lineage (SUBPHYLUM)"
        },
        {
          "id": "tl:subspecies",
          "label": "Taxonomic lineage (SUBSPECIES)"
        },
        {
          "id": "tl:subtribe",
          "label": "Taxonomic lineage (SUBTRIBE)"
        },
        {
          "id": "tl:superclass",
          "label": "Taxonomic lineage (SUPERCLASS)"
        },
        {
          "id": "tl:superfamily",
          "label": "Taxonomic lineage (SUPERFAMILY)"
        },
        {
          "id": "tl:superkingdom",
          "label": "Taxonomic lineage (SUPERKINGDOM)"
        },
        {
          "id": "tl:superorder",
          "label": "Taxonomic lineage (SUPERORDER)"
        },
        {
          "id": "tl:superphylum",
          "label": "Taxonomic lineage (SUPERPHYLUM)"
        },
        {
          "id": "tl:tribe",
          "label": "Taxonomic lineage (TRIBE)"
        },
        {
          "id": "tl:varietas",
          "label": "Taxonomic lineage (VARIETAS)"
        }
      ]
    }
  ],
  "links": [
    {
      "id": "2D gel",
      "title": "2D gel",
      "items": [
        {
          "id": "dr:compluyeast-2dpage",
          "label": "COMPLUYEAST-2DPAGE"
        },
        {
          "id": "dr:dosac-cobs-2dpage",
          "label": "DOSAC-COBS-2DPAGE"
        },
        {
          "id": "dr:ogp",
          "label": "OGP"
        },
        {
          "id": "dr:reproduction-2dpage",
          "label": "REPRODUCTION-2DPAGE"
        },
        {
          "id": "dr:swiss-2dpage",
          "label": "SWISS-2DPAGE"
        },
        {
          "id": "dr:ucd-2dpage",
          "label": "UCD-2DPAGE"
        },
        {
          "id": "dr:world-2dpage",
          "label": "World-2DPAGE"
        }
      ]
    },
    {
      "id": "3D structure",
      "title": "3D structure",
      "items": [
        {
          "id": "dr:pdb",
          "label": "PDB"
        },
        {
          "id": "dr:pdbsum",
          "label": "PDBsum"
        },
        {
          "id": "dr:smr",
          "label": "SMR"
        }
      ]
    },
    {
      "id": "Chemistry",
      "title": "Chemistry",
      "items": [
        {
          "id": "dr:bindingdb",
          "label": "BindingDB"
        },
        {
          "id": "dr:chembl",
          "label": "ChEMBL"
        },
        {
          "id": "dr:drugbank",
          "label": "DrugBank"
        },
        {
          "id": "dr:guidetopharmacology",
          "label": "GuidetoPHARMACOLOGY"
        },
        {
          "id": "dr:swisslipids",
          "label": "SwissLipids"
        }
      ]
    },
    {
      "id": "Enzyme and pathway",
      "title": "Enzyme and pathway",
      "items": [
        {
          "id": "dr:biocyc",
          "label": "BioCyc"
        },
        {
          "id": "dr:brenda",
          "label": "BRENDA"
        },
        {
          "id": "dr:reactome",
          "label": "Reactome"
        },
        {
          "id": "dr:sabio-rk",
          "label": "SABIO-RK"
        },
        {
          "id": "dr:signalink",
          "label": "SignaLink"
        },
        {
          "id": "dr:signor",
          "label": "SIGNOR"
        },
        {
          "id": "dr:unipathway",
          "label": "UniPathway"
        },
        {
          "id": "dr:plantreactome",
          "label": "PlantReactome"
        }
      ]
    },
    {
      "id": "Family and domain",
      "title": "Family and domain",
      "items": [
        {
          "id": "dr:disprot",
          "label": "DisProt"
        },
        {
          "id": "dr:cdd",
          "label": "CDD"
        },
        {
          "id": "dr:gene3d",
          "label": "Gene3D"
        },
        {
          "id": "dr:hamap",
          "label": "HAMAP"
        },
        {
          "id": "dr:interpro",
          "label": "InterPro"
        },
        {
          "id": "dr:panther",
          "label": "PANTHER"
        },
        {
          "id": "dr:pfam",
          "label": "Pfam"
        },
        {
          "id": "dr:pirsf",
          "label": "PIRSF"
        },
        {
          "id": "dr:prints",
          "label": "PRINTS"
        },
        {
          "id": "dr:prodom",
          "label": "ProDom"
        },
        {
          "id": "dr:sfld",
          "label": "SFLD"
        },
        {
          "id": "dr:smart",
          "label": "SMART"
        },
        {
          "id": "dr:supfam",
          "label": "SUPFAM"
        },
        {
          "id": "dr:tigrfams",
          "label": "TIGRFAMs"
        },
        {
          "id": "dr:prosite",
          "label": "PROSITE"
        }
      ]
    },
    {
      "id": "Gene expression",
      "title": "Gene expression",
      "items": [
        {
          "id": "dr:bgee",
          "label": "Bgee"
        },
        {
          "id": "dr:cleanex",
          "label": "CleanEx"
        },
        {
          "id": "dr:collectf",
          "label": "CollecTF"
        },
        {
          "id": "dr:expressionatlas",
          "label": "ExpressionAtlas"
        },
        {
          "id": "dr:genevisible",
          "label": "Genevisible"
        }
      ]
    },
    {
      "id": "Genome annotation",
      "title": "Genome annotation",
      "items": [
        {
          "id": "dr:ensembl",
          "label": "Ensembl"
        },
        {
          "id": "dr:ensemblbacteria",
          "label": "EnsemblBacteria"
        },
        {
          "id": "dr:ensemblfungi",
          "label": "EnsemblFungi"
        },
        {
          "id": "dr:ensemblmetazoa",
          "label": "EnsemblMetazoa"
        },
        {
          "id": "dr:ensemblplants",
          "label": "EnsemblPlants"
        },
        {
          "id": "dr:ensemblprotists",
          "label": "EnsemblProtists"
        },
        {
          "id": "dr:genedb",
          "label": "GeneDB"
        },
        {
          "id": "dr:geneid",
          "label": "GeneID"
        },
        {
          "id": "dr:gramene",
          "label": "Gramene"
        },
        {
          "id": "dr:kegg",
          "label": "KEGG"
        },
        {
          "id": "dr:patric",
          "label": "PATRIC"
        },
        {
          "id": "dr:ucsc",
          "label": "UCSC"
        },
        {
          "id": "dr:vectorbase",
          "label": "VectorBase"
        },
        {
          "id": "dr:wbparasite",
          "label": "WBParaSite"
        }
      ]
    },
    {
      "id": "Ontologies",
      "title": "Ontologies",
      "items": [
        {
          "id": "dr:go",
          "label": "GO"
        }
      ]
    },
    {
      "id": "Organism-specific",
      "title": "Organism-specific",
      "items": [
        {
          "id": "dr:arachnoserver",
          "label": "ArachnoServer"
        },
        {
          "id": "dr:araport",
          "label": "Araport"
        },
        {
          "id": "dr:cgd",
          "label": "CGD"
        },
        {
          "id": "dr:conoserver",
          "label": "ConoServer"
        },
        {
          "id": "dr:ctd",
          "label": "CTD"
        },
        {
          "id": "dr:dictybase",
          "label": "dictyBase"
        },
        {
          "id": "dr:disgenet",
          "label": "DisGeNET"
        },
        {
          "id": "dr:echobase",
          "label": "EchoBASE"
        },
        {
          "id": "dr:ecogene",
          "label": "EcoGene"
        },
        {
          "id": "dr:euhcvdb",
          "label": "euHCVdb"
        },
        {
          "id": "dr:eupathdb",
          "label": "EuPathDB"
        },
        {
          "id": "dr:flybase",
          "label": "FlyBase"
        },
        {
          "id": "dr:genecards",
          "label": "GeneCards"
        },
        {
          "id": "dr:genereviews",
          "label": "GeneReviews"
        },
        {
          "id": "dr:h-invdb",
          "label": "H-InvDB"
        },
        {
          "id": "dr:hgnc",
          "label": "HGNC"
        },
        {
          "id": "dr:hpa",
          "label": "HPA"
        },
        {
          "id": "dr:legiolist",
          "label": "LegioList"
        },
        {
          "id": "dr:leproma",
          "label": "Leproma"
        },
        {
          "id": "dr:maizegdb",
          "label": "MaizeGDB"
        },
        {
          "id": "dr:malacards",
          "label": "MalaCards"
        },
        {
          "id": "dr:mgi",
          "label": "MGI"
        },
        {
          "id": "dr:mim",
          "label": "MIM"
        },
        {
          "id": "dr:niagads",
          "label": "NIAGADS"
        },
        {
          "id": "dr:nextprot",
          "label": "neXtProt"
        },
        {
          "id": "dr:opentargets",
          "label": "OpenTargets"
        },
        {
          "id": "dr:orphanet",
          "label": "Orphanet"
        },
        {
          "id": "dr:pharmgkb",
          "label": "PharmGKB"
        },
        {
          "id": "dr:pombase",
          "label": "PomBase"
        },
        {
          "id": "dr:pseudocap",
          "label": "PseudoCAP"
        },
        {
          "id": "dr:rgd",
          "label": "RGD"
        },
        {
          "id": "dr:sgd",
          "label": "SGD"
        },
        {
          "id": "dr:tair",
          "label": "TAIR"
        },
        {
          "id": "dr:tuberculist",
          "label": "TubercuList"
        },
        {
          "id": "dr:vgnc",
          "label": "VGNC"
        },
        {
          "id": "dr:wormbase",
          "label": "WormBase"
        },
        {
          "id": "dr:xenbase",
          "label": "Xenbase"
        },
        {
          "id": "dr:zfin",
          "label": "ZFIN"
        }
      ]
    },
    {
      "id": "Other",
      "title": "Other",
      "items": [
        {
          "id": "dr:chitars",
          "label": "ChiTaRS"
        },
        {
          "id": "dr:evolutionarytrace",
          "label": "EvolutionaryTrace"
        },
        {
          "id": "dr:genewiki",
          "label": "GeneWiki"
        },
        {
          "id": "dr:genomernai",
          "label": "GenomeRNAi"
        },
        {
          "id": "dr:pmap-cutdb",
          "label": "PMAP-CutDB"
        },
        {
          "id": "dr:pro",
          "label": "PRO"
        },
        {
          "id": "dr:proteomes",
          "label": "Proteomes"
        }
      ]
    },
    {
      "id": "PTM",
      "title": "PTM",
      "items": [
        {
          "id": "dr:carbonyldb",
          "label": "CarbonylDB"
        },
        {
          "id": "dr:depod",
          "label": "DEPOD"
        },
        {
          "id": "dr:glyconnect",
          "label": "GlyConnect"
        },
        {
          "id": "dr:iptmnet",
          "label": "iPTMnet"
        },
        {
          "id": "dr:phosphositeplus",
          "label": "PhosphoSitePlus"
        },
        {
          "id": "dr:swisspalm",
          "label": "SwissPalm"
        },
        {
          "id": "dr:unicarbkb",
          "label": "UniCarbKB"
        }
      ]
    },
    {
      "id": "Phylogenomic",
      "title": "Phylogenomic",
      "items": [
        {
          "id": "dr:eggnog",
          "label": "eggNOG"
        },
        {
          "id": "dr:genetree",
          "label": "GeneTree"
        },
        {
          "id": "dr:hogenom",
          "label": "HOGENOM"
        },
        {
          "id": "dr:inparanoid",
          "label": "InParanoid"
        },
        {
          "id": "dr:ko",
          "label": "KO"
        },
        {
          "id": "dr:oma",
          "label": "OMA"
        },
        {
          "id": "dr:orthodb",
          "label": "OrthoDB"
        },
        {
          "id": "dr:phylomedb",
          "label": "PhylomeDB"
        },
        {
          "id": "dr:treefam",
          "label": "TreeFam"
        }
      ]
    },
    {
      "id": "Polymorphism and mutation",
      "title": "Polymorphism and mutation",
      "items": [
        {
          "id": "dr:biomuta",
          "label": "BioMuta"
        },
        {
          "id": "dr:dmdm",
          "label": "DMDM"
        },
        {
          "id": "dr:dbsnp",
          "label": "dbSNP"
        }
      ]
    },
    {
      "id": "Protein family/group",
      "title": "Protein family/group",
      "items": [
        {
          "id": "dr:allergome",
          "label": "Allergome"
        },
        {
          "id": "dr:cazy",
          "label": "CAZy"
        },
        {
          "id": "dr:esther",
          "label": "ESTHER"
        },
        {
          "id": "dr:imgt_gene-db",
          "label": "IMGT_GENE-DB"
        },
        {
          "id": "dr:merops",
          "label": "MEROPS"
        },
        {
          "id": "dr:moondb",
          "label": "MoonDB"
        },
        {
          "id": "dr:moonprot",
          "label": "MoonProt"
        },
        {
          "id": "dr:mycoclap",
          "label": "mycoCLAP"
        },
        {
          "id": "dr:peroxibase",
          "label": "PeroxiBase"
        },
        {
          "id": "dr:rebase",
          "label": "REBASE"
        },
        {
          "id": "dr:tcdb",
          "label": "TCDB"
        },
        {
          "id": "dr:unilectin",
          "label": "UniLectin"
        }
      ]
    },
    {
      "id": "Protein-protein interaction",
      "title": "Protein-protein interaction",
      "items": [
        {
          "id": "dr:biogrid",
          "label": "BioGrid"
        },
        {
          "id": "dr:complexportal",
          "label": "ComplexPortal"
        },
        {
          "id": "dr:corum",
          "label": "CORUM"
        },
        {
          "id": "dr:dip",
          "label": "DIP"
        },
        {
          "id": "dr:elm",
          "label": "ELM"
        },
        {
          "id": "dr:intact",
          "label": "IntAct"
        },
        {
          "id": "dr:mint",
          "label": "MINT"
        },
        {
          "id": "dr:string",
          "label": "STRING"
        }
      ]
    },
    {
      "id": "Proteomes",
      "title": "Proteomes",
      "items": []
    },
    {
      "id": "Proteomic",
      "title": "Proteomic",
      "items": [
        {
          "id": "dr:cptac",
          "label": "CPTAC"
        },
        {
          "id": "dr:epd",
          "label": "EPD"
        },
        {
          "id": "dr:maxqb",
          "label": "MaxQB"
        },
        {
          "id": "dr:paxdb",
          "label": "PaxDb"
        },
        {
          "id": "dr:peptideatlas",
          "label": "PeptideAtlas"
        },
        {
          "id": "dr:pride",
          "label": "PRIDE"
        },
        {
          "id": "dr:promex",
          "label": "ProMEX"
        },
        {
          "id": "dr:proteomicsdb",
          "label": "ProteomicsDB"
        },
        {
          "id": "dr:topdownproteomics",
          "label": "TopDownProteomics"
        },
        {
          "id": "dr:jpost",
          "label": "jPOST"
        },
        {
          "id": "dr:massive",
          "label": "MassIVE"
        }
      ]
    },
    {
      "id": "Protocols and materials",
      "title": "Protocols and materials",
      "items": [
        {
          "id": "dr:dnasu",
          "label": "DNASU"
        },
        {
          "id": "dr:abcd",
          "label": "ABCD"
        }
      ]
    },
    {
      "id": "Sequence",
      "title": "Sequence",
      "items": [
        {
          "id": "dr:embl",
          "label": "EMBL"
        },
        {
          "id": "dr:ccds",
          "label": "CCDS"
        },
        {
          "id": "dr:pir",
          "label": "PIR"
        },
        {
          "id": "dr:refseq",
          "label": "RefSeq"
        }
      ]
    }
  ]
}
