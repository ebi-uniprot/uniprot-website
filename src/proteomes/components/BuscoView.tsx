import { CSSProperties, useMemo } from 'react';

import { formatPercentage } from '../../shared/utils/utils';

import { BuscoReport } from '../adapters/proteomesConverter';

import styles from './styles/busco-view.module.scss';

interface Style extends CSSProperties {
  '--s': string;
  '--d': string;
  '--f': string;
}

const getPercentageOfTotal = (total: number) => (x: number) =>
  (100 * x) / total;

const BuscoView = ({ report }: { report: BuscoReport }) => {
  const [cssProperties, formattedPercentages] = useMemo(() => {
    const getPercentage = getPercentageOfTotal(report.total);
    const percentages = {
      c: getPercentage(report.complete),
      s: getPercentage(report.completeSingle),
      d: getPercentage(report.completeDuplicated),
      f: getPercentage(report.fragmented),
      m: getPercentage(report.missing),
    };
    const cssProperties: Style = {
      '--s': `${percentages.s}%`,
      '--d': `${percentages.d}%`,
      '--f': `${percentages.f}%`,
    };
    const formattedPercentages = {
      c: formatPercentage(percentages.c),
      s: formatPercentage(percentages.s),
      d: formatPercentage(percentages.d),
      f: formatPercentage(percentages.f),
      m: formatPercentage(percentages.m),
    };
    return [cssProperties, formattedPercentages];
  }, [report]);

  return (
    <div className={styles['busco-view']}>
      <div className={styles['busco-view__track']} style={cssProperties} />
      {/* TODO: add after implementation from backend - the creation date of the dataset (format: 2019-01-04) */}
      <div
        className={styles['busco-view__report']}
      >{`n:${report.total} Â· ${report.lineageDb}`}</div>
      <div>{`C:${formattedPercentages.c} (S:${formattedPercentages.s} D:${formattedPercentages.d}) F:${formattedPercentages.f} M:${formattedPercentages.m}`}</div>
    </div>
  );
};

export default BuscoView;
