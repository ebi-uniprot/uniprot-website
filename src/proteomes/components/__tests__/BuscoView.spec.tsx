import { render } from '@testing-library/react';

import BuscoView from '../BuscoView';

import data from '../../__mocks__/proteomesEntryModelData';

describe('BuscoView', () => {
  it('should render', () => {
    const { asFragment } = render(
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      <BuscoView report={data.proteomeCompletenessReport.buscoReport!} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
