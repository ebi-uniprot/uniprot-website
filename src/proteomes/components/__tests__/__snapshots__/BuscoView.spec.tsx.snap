// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuscoView should render 1`] = `
<DocumentFragment>
  <div
    class="busco-view"
  >
    <protvista-track
      height="10"
      length="100"
    />
    <div
      class="busco-view__report"
    >
      n:13780 Â· primates_odb10
    </div>
    <div>
      C:99.5% (S:37.8% D:61.7%) F:0% M:0.5%
    </div>
  </div>
</DocumentFragment>
`;
